src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,105 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * This class is a subclass of TickingTimer. It contains logic to facilitate dynamic period of the timer.
+ */
+public class RefreshTimer extends TickingTimer{
+
+    /**
+     * The buffer time to be added to ensure refresh happens after the api quota renewal.
+     */
+    public static final int BUFFER_TIME = 1;
```

Comment:
I don't think this should be defined as a constant. It's something internal to the calculation, not something that needs to be visible at class level, right?